sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,267 @@
+"""Stacking API"""
+
+# Author: Caio Oliveira <caioaao@gmail.com>
+# License: BSD 3 clause
+
+from ..base import (BaseEstimator, TransformerMixin, MetaEstimatorMixin)
+from ..model_selection import cross_val_predict
+from ..pipeline import (_name_estimators, FeatureUnion)
+from ..preprocessing import FunctionTransformer
+
+
+class StackMetaEstimator(BaseEstimator, MetaEstimatorMixin, TransformerMixin):
+    """Transformer to turn estimators into meta-estimators for model stacking
+
+    In stacked generalization, meta estimators are combined in layers to
+    improve the final result. To prevent data leaks between layers, a procedure
+    similar to cross validation is adopted, where the model is trained in one
+    part of the set and predicts the other part. In `StackMetaEstimator`, it
+    happens during `fit_transform`, as the result of this procedure is what
+    should be used by the next layers.
+
+    Parameters
+    ----------
+    base_estimator : the estimator to be blended.
+
+    cv : cv to be used, optional (default=3)
+        Will be passed to `cross_val_predict` during `fit_transform`.
+
+    method : string, optional (default='auto')
+        Invokes the passed method name of the passed estimator. If the method
+        is `auto`, will try to invoke `predict_proba` or `predict` in that
```

Comment:
Some estimators support `decision_function` and do not support `predict_proba`. I think that should be in the back-off list of methods.